bcipy/tasks/rsvp/main_frame.py

```diff
@@ -253,3 +261,77 @@ def prepare_stimuli(self):
             timing=self.stimuli_timing,
             seq_constants=self.seq_constants)
         return stimuli
+
+
+class CriteriaEvaluator():
+    """Evaluates whether an epoch should commit to a decision based on the
+    provided criteria.
+    
+    Parameters:
+    -----------
+        continue_criteria: list of functions; if any of these evaluate to true the
+            decision maker continues.
+        commit_criteria: list of functions; if any of these return true and
+            continue_criteria are all false, decision maker commits to a decision.
+    """
+
+    def __init__(self, continue_criteria: list, commit_criteria: list):
+        self.continue_criteria = continue_criteria or []
+        self.commit_criteria = commit_criteria or []
+
+    @classmethod
+    def default(cls):
+        return cls(
+            continue_criteria=[less_than_min_iterations],
+            commit_criteria=[exceeds_max_iterations, meets_commit_threshold])
+
+    def should_commit(self, epoch: Dict, params: Dict):
+        """Evaluates the given epoch; returns true if stoppage criteria has
+        been met, otherwise false.
+
+        Parameters:
+        -----------
+            epoch - Epoch data
+            params - params relevant to stoppage criteria
+                min_num_seq: int - minimum number of sequences required
+                max_num_seq: int - max number of sequences allowed
+                threshold: float - minimum likelihood required
+        """
+        if any(criteria(epoch, params) for criteria in self.continue_criteria):
+            return False
+        return any(
+            criteria(epoch, params) for criteria in self.commit_criteria)
+
+
+# Criteria
+
+def less_than_min_iterations(epoch, commit_params):
```

Comment:
There is a high possibility of needing these criteria outside of RSVP (also the `CriteriaEvaluator `). It may be worth thinking of where we could pull this out too.